app/experimenter/experiments/tests/test_changelog_utils/test_changelog_utils_nimbus.py

```diff
@@ -0,0 +1,162 @@
+from django.test import TestCase
+
+from experimenter.experiments.changelog_utils import (
+    NimbusExperimentChangeLogSerializer,
+    generate_nimbus_changelog,
+)
+from experimenter.experiments.models import NimbusExperiment
+from experimenter.experiments.tests.factories import (
+    NimbusExperimentFactory,
+    NimbusProbeSetFactory,
+)
+from experimenter.experiments.tests.factories.nimbus import NimbusFeatureConfigFactory
+from experimenter.openidc.tests.factories import UserFactory
+from experimenter.projects.tests.factories import ProjectFactory
+
+
+class TestNimbusExperimentChangeLogSerializer(TestCase):
+    maxDiff = None
+
+    def test_outputs_expected_schema_for_empty_experiment(self):
+        experiment = NimbusExperiment()
+        data = dict(NimbusExperimentChangeLogSerializer(experiment).data)
+        self.assertEqual(
+            data,
+            {
+                "application": None,
+                "branches": [],
+                "channels": [],
+                "control_branch": None,
+                "feature_config": None,
+                "firefox_min_version": None,
+                "hypothesis": NimbusExperiment.HYPOTHESIS_DEFAULT,
+                "is_paused": False,
+                "name": "",
+                "owner": None,
+                "population_percent": "0.0000",
+                "probe_sets": [],
+                "projects": [],
+                "proposed_duration": None,
+                "proposed_enrollment": None,
+                "public_description": None,
+                "slug": "",
+                "status": NimbusExperiment.Status.DRAFT.value,
+                "targeting_config_slug": None,
+            },
+        )
+
+    def test_outputs_expected_schema_for_complete_experiment(self):
+        application = NimbusExperiment.Application.DESKTOP
+        probe_set = NimbusProbeSetFactory.create()
+        feature_config = NimbusFeatureConfigFactory.create()
+        project = ProjectFactory.create()
+
+        experiment = NimbusExperimentFactory.create_with_status(
+            NimbusExperiment.Status.COMPLETE,
+            application=application,
+            probe_sets=[probe_set],
+            feature_config=feature_config,
+            projects=[project],
+        )
+        data = dict(NimbusExperimentChangeLogSerializer(experiment).data)
```

Comment:
feel like i'm missing something, why do we need to cast it to a dict?

Reply:
It comes out of the serializer as an OrderedDict, so for proper dict comparisons gotta cast it to a dict explicitly

Reply:
In some of our older tests we did it the other way around and compared everything as OrderedDicts but when I was playing with it I found it simpler to compare everything as dicts

Reply:
ohhhhh right, the pain of OrderedDicts. Makes sense